server/py/services/api/crud/model_monitoring/deployment.py

```
@@ -741,16 +750,203 @@ def _create_tsdb_tables(
+        In addition, it returns the base period of the controller function.
+
+        :param include_infra: If True, include the model monitoring infrastructure functions in the response.
+
+        :return: Returns a tuple containing a list of FunctionSummary objects and the base period of the controller
+                 function. If `include_infra` is False, the list will be empty.
+
+        """
+        function_summaries_list = []
+        base_period = 0
+        if include_infra:
+            infra_mm_functions = self.list_model_monitoring_functions(
+                format_=mlrun.common.formatters.FunctionFormat.full, infra_only=True
+            )
+
+            if not infra_mm_functions:
+                logger.info("No model monitoring infrastructure functions found")
+            for function in infra_mm_functions:
+                function_summary = (
+                    mlrun.common.schemas.model_monitoring.FunctionSummary.from_dict(
+                        function, func_type="infra"
+                    )
+                )
+                function_summaries_list.append(function_summary)
+                if (
+                    function["metadata"]["name"]
+                    == mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER
+                ):
+                    base_period = self._get_base_period(controller_func=function)
+        else:
+            # getting the base period from the controller function
+            try:
+                controller_func = services.api.crud.Functions().get_function(
+                    db_session=self.db_session,
+                    name=mm_constants.MonitoringFunctionNames.APPLICATION_CONTROLLER,
+                    project=self.project,
+                )
+                base_period = self._get_base_period(controller_func=controller_func)
+            except mlrun.errors.MLRunNotFoundError:
+                logger.info(
```

Are we sure we want to log it and ignore this error?

---

yes, because this isn’t a real issue. If the user disables mm after data already exists, we remove the controller func, even if application functions remain. In that case, there is no `base period` but the missing controller isn’t an error, it’s expected behavior.